core/headerchain.go

```diff
@@ -1374,6 +1374,11 @@ func (hc *HeaderChain) WriteUtxoViewpoint(view *types.UtxoViewpoint) error {
 
 		fmt.Println("write utxo", outpoint.Hash.Hex(), outpoint.Index)
 		rawdb.WriteUtxo(hc.bc.db, outpoint.Hash, outpoint.Index, entry)
+		entryAddress := common.BytesToAddress(entry.Address, hc.NodeLocation())
+		addressUtxos := rawdb.ReadAddressUtxos(hc.bc.db, entryAddress)
+		addressUtxos = append(addressUtxos, entry)
+		fmt.Println("utxos for address", len(addressUtxos))
+		rawdb.WriteAddressUtxos(hc.bc.db, entryAddress, addressUtxos)
```

Comment:
You’re effectively storing the entire UTXO set twice here. Because the set can potentially be billions of UTXOs for each zone (and therefore 100gb+) I don’t think this is acceptable.